File: src/main/java/com/graqr/threshr/model/redsky/product/PrimaryBrand.java
package com.graqr.threshr.model.redsky.product;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.serde.annotation.Serdeable;

@Serdeable
public record PrimaryBrand(
        @JsonProperty("canonical_url") String canonicalUrl,
        @JsonProperty("linking_id") String linkingId,
        String name) {
}

File: src/main/kotlin/com/graqr/threshr/circleStore/model/product/PrimaryBrand.kt
package com.graqr.threshr.circleStore.model.product

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class PrimaryBrand(
    @param:JsonProperty("canonical_url") val canonicalUrl: String,
    @param:JsonProperty("linking_id") val linkingId: String,
    val name: String
) {}
